File: python/examples/image_generation_example.py
import json
import base64
from pathlib import Path
from aime_api_client_interface import do_api_request

def generate_image():
    # Define the image generation parameters
    params = {
        'prompt': 'Astronaut on Mars holding a banner which states "AIME is happy to serve your model" during sunset sitting on a giant yellow rubber duck',
        'seed': -1,
        'height': 1024,
        'width': 1024,
        'steps': 50,
        'guidance': 3.5,
        'image2image_strength': 0.8,
        'provide_progress_images': 'none',
        'wait_for_result': True
    }

    # Call the AIME API
    final = do_api_request(
        'https://api.aime.info',
        'flux-dev',
        params,
        user='apiexample@aime.info',
        key='181e35ac-7b7d-4bfe-9f12-153757ec3952'
    )

    # Save the images
    images = final.get('images') or final.get('job_result', {}).get('images', [])
    if not images:
        print("No images returned by the API.")
        return final
    for i, img_b64 in enumerate(images):
        header, img_data = img_b64.split(',', 1) if ',' in img_b64 else (None, img_b64)
        img_bytes = base64.b64decode(img_data)
        filename = Path(__file__).parent / f'image_{i}.png'
        filename.write_bytes(img_bytes)
        print(f"Saved image to: {filename}")
    print(f"\nImage generation complete. {len(images)} image(s) saved.")
    return final

if __name__ == "__main__":
    generate_image()

File: js/examples/image_generation_example.js
const { doAPIRequest } = require('../model_api');
const fs = require('fs');

// Configuration
const CONFIG = {
    BASE_URL: 'https://api.aime.info',
    ENDPOINT: 'flux-dev',
    API_EMAIL: 'apiexample@aime.info',
    API_KEY: '181e35ac-7b7d-4bfe-9f12-153757ec3952'
};

// Set up global fetch to handle base URL
const { fetch: originalFetch } = global;
global.fetch = (input, init = {}) => {
    const url = typeof input === 'string' 
        ? input.startsWith('http') ? input : `${CONFIG.BASE_URL}${input.startsWith('/') ? '' : '/'}${input}`
        : input;
    return originalFetch(url, init);
};

// Image generation parameters
const params = {
    prompt: 'A beautiful sunset over mountains',
    height: 1024,
    width: 1024,
    steps: 30,
    guidance: 7.5,
    wait_for_result: true
};

console.log('🚀 Starting image generation...');

// Make the API request
doAPIRequest(
    CONFIG.ENDPOINT,
    params,
    (result, error) => {
        if (error) {
            console.error('❌ Error:', error.message || 'Unknown error');
            return;
        }

        const images = result.images || [];
        if (images.length === 0) {
            console.log('⚠️  No images were generated');
            return;
        }

        // Save the first image
        const imageData = images[0].includes(',') ? images[0].split(',')[1] : images[0];
        const outputPath = 'generated_image.png';
        fs.writeFileSync(outputPath, Buffer.from(imageData, 'base64'));
        
        console.log(`✅ Image saved as ${outputPath}`);
        console.log('✨ All done!');
    },
    CONFIG.API_EMAIL,
    CONFIG.API_KEY,
    (progress) => {
        const msg = progress.progress >= 0 
            ? `Progress: ${progress.progress}%` 
            : 'Starting...';
        process.stdout.write(`\r${msg}${' '.repeat(20)}`);
    }
);
